internal/chart/helm_client.go

```diff
@@ -51,9 +61,75 @@ func getActionConfig(namespace string) (*action.Configuration, error) {
 	return actionConfig, nil
 }
 
-// Option to perform additional configuration of action.Install before running a chart installation.
+// InstallOption to perform additional configuration of action.Install before running a chart installation.
 type InstallOption func(install *action.Install)
 
+type postRender struct {
+	client.Client
+	namespace string
+}
+
+func (p *postRender) Run(renderedManifests *bytes.Buffer) (modifiedManifests *bytes.Buffer, err error) {
+
+	var configMapList v1.ConfigMapList
+	if err := p.Client.List(context.Background(), &configMapList); err != nil {
+		return nil, err
+	}
+
+	fs := filesys.MakeFsInMemory()
+	if err := fs.Mkdir(p.namespace); err != nil {
+		return nil, err
+	}
+
+	var postrenderFound bool
+	for _, cm := range configMapList.Items {
+		// only apply the postrenders present in the current namespace
+		if cm.Namespace == p.namespace {
```

Comment:
I think it would be nice if we could try to decrease quantity of all of these nested statements.
```golang
	var postrenderFound bool
	for _, cm := range configMapList.Items {
		if !strings.HasSuffix(cm.Name, "-postrender") {
			continue
		}
		// check if a postrender is available
		postrenderFound = true
		for k, v := range cm.Data {
			fileName := p.namespace + "/" + k
			if err := fs.WriteFile(fileName, []byte(v)); err != nil {
				return nil, err
			}
		}
	}


```

Reply:
Changed.